### ISSUE TITLE
Open the main app from floating window?

### ISSUE THREAD

**Issue by ducviet321** [X-SLAYER/flutter_overlay_window#108]
I'm looking for a better solution to open the main application from the floating window, right now I'm only able to archive this by modifying `OverlayServices.java` and call it as `FlutterOverlayWindow.openMainApp();`

```java
    private void openMainApp() {
        Intent intent = new Intent();
        intent.setClassName("com.vitinc.myapp", "com.vitinc.myapp.MainActivity");
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }
```

**Comment by macdo-py** [comment]
Hi @ducviet321 , Im very interested in your idea / implementation, could not make it work, could you please explain a little more.
The file you are referencing is: OverlayService.java?
In wich line did you add your code?
Do you need to do something else beside calling it as you said?

Thanks in advance!

**Comment by ducviet321** [comment]
In `/lib/src/overlay_window.dart` add to `class FlutterOverlayWindow`

```dart
static Future<bool?> openMainApp() async {
    final bool? _res = await _overlayChannel.invokeMethod<bool?>('openMainApp');
    return _res;
  }
```

in `android/src/main/java/flutter/overlay/window/flutter_overlay_window/OverlayService.java`

Find `flutterChannel.setMethodCallHandler((call, result) -> {` and add below
```dart
            } else if (call.method.equals("resizeOverlay")) {
               ...
            } else if (call.method.equals("openMainApp")) {
                Intent intent = new Intent();
                intent.setClassName("com.vitinc.widpet_android", "com.vitinc.widpet_android.MainActivity");
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);
            }
```

**Comment by macdo-py** [comment]
@ducviet321  thank you so much for your help!!! it worked but I hag to remove the line
`Intent intent = new Intent();`
And also, I'd like to know why you added a space in your app namespace, was it a typo o it should be like that.

Thanks again and best regards!

**Comment by ducviet321** [comment]
Yeah it was my typo, maybe be you should rename intent variable instead of removing it, I haven't looked, just try to make it work quickly

**Comment by icanall10** [comment]
use https://pub.dev/packages/android_intent_plus with app links

**Comment by selvam920** [comment]
@icanall10 can you share example to open the app using android_intent_plus?

**Comment by icanall10** [comment]
`
AndroidIntent intent = AndroidIntent(
      action: 'action_view',
      data: 'your-app-link-schema://your-app-link-host/some-path',
);

await intent.launch();
`

**Comment by icanall10** [comment]
@selvam920 
 `
AndroidIntent intent = AndroidIntent(
      action: 'action_view',
      data: 'your-app-link-schema://your-app-link-host/some-path',
);

await intent.launch();
`

**Comment by selvam920** [comment]
thanks

**Comment by notTural** [comment]
I have added this functionality in [this](https://github.com/X-SLAYER/flutter_overlay_window/pull/180) pr that dynamically calls the app without hard coding app namespace like @ducviet321 proposed.

**Comment by notTural** [comment]
> [@selvam920](https://github.com/selvam920) you should pre-configure app_links and then run
> 
> ```
> AndroidIntent intent = AndroidIntent(
>       action: 'action_view',
>       data: 'your-app-link-schema://your-app-link-host/some-path',
> );
> 
> await intent.launch();
> ```

When I tried this method, the application was not resumed from where it was left. It was as if the application was closed and relaunched.
